### PR TITLE
User-defined time zone

### LINKED ISSUE #2754: Add Time zone options for each user
Currently, only time zone settings in php.ini can be used, which is very bad, and you should be able to set the global time zone at config.php, as well as each user setting the time zone separately.

### PR DESCRIPTION
**Pull request by Alkarex**
#fix https://github.com/FreshRSS/FreshRSS/issues/2754

*Default* is server / PHP default, e.g. as defined by `TZ` in our Docker images, or in `php.ini` `date.timezone`

image

### CODE REVIEW

**Comment by Alkarex** [comment]
image

File: app/i18n/nl/conf.php [hunk 1]
```diff
@@ -57,6 +57,7 @@
 			'portrait' => 'Staand',
 			'square' => 'Vierkant',
 		),
+		'timezone' => 'Time zone',	// TODO
```
Comment (Frenzie) [inline app/i18n/nl/conf.php]:
```suggestion
		'timezone' => 'Tijdzone',
```

**Review by Frenzie — APPROVED** [review]
Looks straightforward, can't think of any negatives.

**Comment by aledeg** [comment]
Correct me if I'm wrong, but for what I can gather from the code, there is no use of the timezone elsewhere than in the configuration. Do you plan on adding some code to utilize it in the interface or am I missing something?

File: app/FreshRSS.php [hunk 2]
```diff
@@ -99,6 +99,9 @@ private static function initI18n() {
 
 		Minz_Session::_param('language', $language);
 		Minz_Translate::init($language);
+
+		$timezone = isset(FreshRSS_Context::$user_conf) ? FreshRSS_Context::$user_conf->timezone : '';
+		date_default_timezone_set($timezone != '' ? $timezone : ini_get('date.timezone'));
```
Comment (Alkarex) [inline app/FreshRSS.php]: @aledeg See here for the actual use

**Comment by Alkarex** [comment]
> Correct me if I'm wrong, but for what I can gather from the code, there is no use of the timezone elsewhere than in the configuration. Do you plan on adding some code to utilize it in the interface or am I missing something?

See the `date_default_timezone_set()` line for the actual use. It changes the time representations in the UI

File: app/FreshRSS.php (hunk 2 above)
Comment (aledeg) [reply app/FreshRSS.php]: Thank you. I missed that part.

**Review by aledeg — APPROVED** [review]

**Comment by Alkarex** [comment]
I have just added the possibility to pick the timezone when creating a new user:

image

**Comment by Alwaysin** [comment]
@Alkarex I think this PR led to a notice in PHP at each page load:
```
AH01071: Got error 'PHP message: PHP Notice:  date_default_timezone_set(): Timezone ID '' is invalid in /var/www/html/freshrss/app/FreshRSS.php on line 104'

```
Is this normal? I guess I just have to set a timezone in my PHP conf which has none by default, or select one in FreshRSS conf?

**Comment by Alkarex** [comment]
@Alwaysin Please try https://github.com/FreshRSS/FreshRSS/pull/5021

**Pull request merged**